libgalois/include/katana/analytics/ClusteringImplementationBase.h

```diff
@@ -442,61 +442,27 @@ struct ClusteringImplementationBase {
  * (read from the underlying RDG) to the in-memory
  * temporary graph (pfg_to).
  */
-  katana::Result<void> CreateDuplicateGraph(
-      katana::PropertyGraph* pfg_from, katana::PropertyGraph* pfg_to,
-      const std::string& edge_property_name,
+  template <typename NodeData>
+  katana::Result<std::unique_ptr<katana::PropertyGraph>> CreateDuplicateGraph(
```

Comment:
nit: why wouldn't we just call this function "DuplicateGraph"?

also, if this takes a template parameter, wouldn't it make more sense to return a TypedPropertyGraph instead?

Reply:
Agree with renaming the function. Not sure if returning a TypedPropertyGraph is worth the change in the code. The call to Duplicate is inside an if-else, each of which produce a PropertyGraph pointer. While typed property graph is constructed afterwards in an iterative loop